src/libraries/System.Linq/src/System/Linq/ElementAt.cs

```diff
@@ -91,5 +172,76 @@ public static TSource ElementAt<TSource>(this IEnumerable<TSource> source, int i
 
             return default;
         }
+
+        /// <summary>Returns the element at a specified index in a sequence or a default value if the index is out of range.</summary>
+        /// <param name="source">An <see cref="IEnumerable{T}" /> to return an element from.</param>
+        /// <param name="index">The index of the element to retrieve, which is either from the start or the end.</param>
+        /// <typeparam name="TSource">The type of the elements of <paramref name="source" />.</typeparam>
+        /// <exception cref="ArgumentNullException">
+        ///   <paramref name="source" /> is <see langword="null" />.
+        /// </exception>
+        /// <returns>
+        ///   <see langword="default" /> if <paramref name="index" /> is outside the bounds of the <paramref name="source" /> sequence; otherwise, the element at the specified position in the <paramref name="source" /> sequence.
+        /// </returns>
+        public static TSource? ElementAtOrDefault<TSource>(this IEnumerable<TSource> source, Index index)
+        {
+            if (source == null)
```

Comment:
There's a lot of duplication between this method and ElementAt.  Can you instead combine them into a private TryGetElementAt?  Then the new public ElementAt would just call it and throw if it returns false, and the new public ElementAtOrDefault would call it and return default if it returns false.

Reply:
Updated with `TryGetElementAt`.